src/Lecture1.hs

```diff
@@ -98,7 +113,13 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> String -> String
+subString _ _ [] = []
+subString start end (x:xs)
+                       | end < 0    = []
+                       | start < 0  = subString 0 end (x:xs)
+                       | start > 0  = subString (start - 1) (end - 1) xs
```

Comment:
That's nice recursive implementation! You can write it shorter using standard functions `drop` and `take` 👍🏻